src/events.rs

```diff
@@ -251,3 +282,10 @@ describe! events_functionality {
         assert_eq!(body, formatted_body);
     }
 }
+#[cfg(test)]
+fn floating_point_events() -> String{
+        let floating_point_string = format!("v=1&t=event&tid=UA-77033033-1&cid=9eccb690-93aa-4513-835a-9a4f0f0e2a71&ec=category&ea=action\
+                                &el=label&ev=1&an=iot_app&av=1.0&ul=en-us&cd1=linux&cd2=1.2&cd3=RPi%2F2&cd4=arm&cd5=rust%20test&cd6=20160320123456\
+                                &cd7=2016-05-25%2022:36:57&cd8=1\n");
+        floating_point_string
```

Comment:
It would be good to create another unit test, copied from `it "should format the body correctly for one event"`.  After you copy that unit test, you can replace the "formatted_body" with your "floating_point_string". "cd8=1" would be better as "cd8=1.0" since "cd8" is the floating point value. Then you can change:

ev.insert_event("category", "action", "label", 1);

to 

ev.insert_event("category", "action", "label", 1.0);

And of course you can rename "should format the body correctly for one event" to "should format the body correctly for one floating point event"